glambie/util/timeseries_helpers.py

```diff
@@ -0,0 +1,358 @@
+                match_arr2 = np.zeros((np.count_nonzero(match_arr1)), dtype=np.int32)
+            else:
+                match_arr2 = np.array([])
+            match_arr1 = np.flatnonzero(match_arr1)
+        return match_arr1, match_arr2
+    concat_arr = np.concatenate((array1, array2))
+    ind = np.concatenate(
+        [np.arange(length_arr1), np.arange(length_arr2)]
+    )
+    vec = np.concatenate(
+        [np.zeros([length_arr1], dtype=bool),
+         np.ones([length_arr2], dtype=bool)]
+    )
+
+    sub = np.argsort(concat_arr)
+    concat_arr = concat_arr[sub]
+    ind = ind[sub]
+    vec = vec[sub]
+
+    if tolerance == 0:
+        firstdup = np.logical_and(
+            concat_arr == np.roll(concat_arr, -1),
+            vec != np.roll(vec, -1)
+        )
+    else:
+        firstdup = np.logical_and(
+            np.abs(concat_arr - np.roll(concat_arr, -1)) < tolerance,
+            vec != np.roll(vec, -1)
+        )
+    count = np.count_nonzero(firstdup)
+    firstdup = np.flatnonzero(firstdup)
+    if count == 0:
+        match_arr1 = np.array([])
+        match_arr2 = np.array([])
+        return match_arr1, match_arr2
+    dup = np.zeros((count * 2), dtype=int)
+    even = np.arange(count) * 2
+
+    dup[even] = firstdup
+    dup[even + 1] = firstdup + 1
+
+    ind = ind[dup]
+    vec = vec[dup]
+    match_arr2 = ind[vec]
+    match_arr1 = ind[np.logical_not(vec)]
+    return match_arr1, match_arr2
+
+
+def resample_1d_array(x: np.array, y: np.array, x_new: np.array, mode: str = "linear") -> np.array:
+    """Simple resampling of a an array: returns interpolated y-values based on a new x-array
+
+    Parameters
+    ----------
+    x : np.array
+        The x-values of the input sequence (e.g an array of fractional dates)
+    y : np.array
+        The y-values of the input sequence
+    x_new : np.array
+        The desired x-values of the output sequence (e.g. an array of the new desired time resolution)
+    mode : str, optional
+        the method of interpolation to use, either 'linear', 'nearest' or 'spline'
+        by default "linear"
+
+    Returns
+    -------
+    np.array
+        And array of the y-values at each value in 'x_new'. Will have the same length as x_new
+
+    Raises
+    ------
+    ValueError
+        if mode not recognised
+    """
+    if mode == "spline":
+        s = interpolate.InterpolatedUnivariateSpline(x, y)
+        ynew = s(x_new)
+    elif mode == "linear":
+        ynew = np.interp(x_new, x, y)
+    elif mode == "nearest":
+        s = interpolate.interp1d(x, y, kind='nearest', fill_value="extrapolate")
+        ynew = s(x_new)
+    else:
+        raise ValueError("Unrecognised interpolation mode")
+
+    return ynew
+
+
+def moving_average(dx: float, x: np.array, y: np.array = None, clip: bool = False) -> np.array:
+    """Calculates the moving average of an array.
+
+    If an argument for y is not provided, the function instead provides a moving average of the signal x,
```

Comment:
This is a little confusing - I think the averaging window is actually 2D and extends over all of y but only dx over x? Can you explain further in the docstring please?

Reply:
Indeed, I changed it so that y is not optional anymore and a moving average has to be calculated over two arrays